src/Microsoft.AspNetCore.Authorization/DefaultAuthorizationService.cs

```diff
@@ -51,16 +52,45 @@ public DefaultAuthorizationService(IAuthorizationPolicyProvider policyProvider,
 
             if (authContext.HasSucceeded)
             {
-                _logger.UserAuthorizationSucceeded(user?.Identity?.Name);
+                _logger.UserAuthorizationSucceeded(GetUserNameForLogging(user));
                 return true;
             }
             else
             {
-                _logger.UserAuthorizationFailed(user?.Identity?.Name);
+                _logger.UserAuthorizationFailed(GetUserNameForLogging(user));
                 return false;
             }
         }
 
+        private string GetUserNameForLogging(ClaimsPrincipal user)
+        {
+            var identity = user?.Identity;
+            if (identity != null)
+            {
+                var name = identity.Name;
+                if (name != null)
+                {
+                    return name;
+                }
+                return GetClaimValue(identity, "sub")
+                    ?? GetClaimValue(identity, ClaimTypes.Name)
+                    ?? GetClaimValue(identity, ClaimTypes.NameIdentifier)
+                    ?? GetClaimValue(identity, ClaimTypes.WindowsAccountName)
```

Comment:
I totally agree with @HaoK, this list is insane and not every claim is accurate for a "username" (e.g the JWT-specific `sub` claim shouldn't be used as a display name, unless you want to end up with random strings) :+1: 

IMHO, we should only use `IIdentity.Name` here and let the identity implementation decide what's appropriate for a username (`ClaimsIdentity` checks for a claim corresponding to `NameClaimType`).